base/floatfuncs.jl

```diff
@@ -278,9 +278,14 @@ true
 function isapprox(x::Number, y::Number;
                   atol::Real=0, rtol::Real=rtoldefault(x,y,atol),
                   nans::Bool=false, norm::Function=abs)
-    x == y || (isfinite(x) && isfinite(y) && norm(x-y) <= max(atol, rtol*max(norm(x), norm(y)))) || (nans && isnan(x) && isnan(y))
+    x == y || (isfinite(x) && isfinite(y) && _isapprox_small(norm(x-y), atol, rtol, x, y, norm)) || (nans && isnan(x) && isnan(y))
 end
 
+# check if d is small compared to atol and rtol, ignoring the units (if any) of atol
+# only if atol is zero.
```

Comment:
It should be possible to pass an `atol` with units.

Right now, `atol` is defined as `::Real`, which doesn't work if the dimensionful type is a subtype of `Number`.   We can change this (and throw an exception if `!isreal(atol)`), or should we assume that dimensionful types are `Real` (so that you make e.g. `Complex{Dimensionful}` rather than `Dimensionful{Complex}`)?

Reply:
Changing `atol` to `Number` and checking throwing if `!isreal(atol)` seems right to me.

Reply:
Done.